logging/logger.go

```diff
@@ -12,179 +12,179 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package logging sets up and configures standard logging.
+// Package logging is an opinionated structured logging library based on
+// [log/slog].
+//
+// This package also aliases most top-level functions in [log/slog] to reduce
+// the need to manage the additional import.
 package logging
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"math"
 	"os"
+	"strconv"
 	"strings"
 	"sync"
-	"time"
 
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
-	"go.uber.org/zap/zaptest"
+	"log/slog"
+
+	"github.com/abcxyz/pkg/timeutil"
 )
 
 // contextKey is a private string type to prevent collisions in the context map.
 type contextKey string
 
-const (
-	// loggerKey points to the value in the context where the logger is stored.
-	loggerKey = contextKey("logger")
-)
+// loggerKey points to the value in the context where the logger is stored.
+const loggerKey = contextKey("logger")
 
-var (
-	// defaultLogger is the default logger. It is initialized once per package
-	// include upon calling DefaultLogger.
-	defaultLogger     *zap.SugaredLogger
-	defaultLoggerOnce sync.Once
-)
+// defaultLogger returns a function that returns the default logger. which
+// writes JSON output to stdout at the "Info" level.
+//
+// It is initialized once when called the first time.
+var defaultLoggerOnce = sync.OnceValue[*slog.Logger](func() *slog.Logger {
+	v, _ := New(os.Stdout, LevelInfo, FormatJSON, false)
```

Comment:
Should we use `NewFromEnv` for the default? One can simply use env vars to set up the default logger.

Reply:
I think it was a mistake that the default logger parsed the environment. `NewFromEnv` can also panic when given invalid input, which seems like a bad idea for the default